src/services/handbrake.service.js

```
@@ -0,0 +1,505 @@
+
+    // Escape shell-sensitive characters for safe shell execution
+    return sanitized.replace(/\\/g, '\\\\').replace(/"/g, '\\"');
+  }
+
+  /**
+   * Builds the HandBrake command with proper arguments
+   * @param {string} handBrakePath - Path to HandBrakeCLI executable
+   * @param {string} inputPath - Path to input MKV file
+   * @param {string} outputPath - Path to output file
+   * @param {string|null} presetOverride - Optional preset override (for retries)
+   * @returns {string} Constructed command
+   * @throws {HandBrakeError} If paths contain invalid characters
+   * @private
+   */
+  static buildCommand(handBrakePath, inputPath, outputPath, presetOverride = null) {
+    const config = AppConfig.handbrake;
+    const preset = presetOverride || config.preset;
+
+    // Validate and sanitize paths
+    if (!handBrakePath || !inputPath || !outputPath) {
+      throw new HandBrakeError('All paths must be provided for HandBrake command');
+    }
+
+    // Sanitize paths to prevent injection
+    const sanitizedHandBrakePath = this.sanitizePath(handBrakePath);
+    const sanitizedInputPath = this.sanitizePath(inputPath);
+    const sanitizedOutputPath = this.sanitizePath(outputPath);
+
+    // Base arguments with proper escaping
+    const args = [
+      `"${sanitizedHandBrakePath}"`,
+      `--input "${sanitizedInputPath}"`,
+      `--output "${sanitizedOutputPath}"`,
+      `--preset "${preset}"`,
+      '--verbose=1', // Enable progress output
+      '--no-dvdnav'  // Disable DVD navigation for better compatibility
+    ];
+
+    // Add format-specific optimizations
+    if (config.output_format.toLowerCase() === 'mp4') {
+      args.push('--optimize');
+    }
+
+    // Add custom arguments if specified (with validation)
+    if (config.additional_args && config.additional_args.trim()) {
+      // Validate additional args don't contain dangerous characters
+      if (/[;&|`$()]/.test(config.additional_args)) {
```

A few things I noticed about the HandBrake argument handling

**Issues:**
- Silent failure (warning only) means users won't know their args were ignored
- The regex doesn't catch all dangerous patterns (e.g., >, <, \n, etc.)

**Recommendation:**
- Throw an error instead of just warning
- Use child_process.spawn with array arguments instead of exec with a shell string

---

Do the updates resolve your concerns?  They were AI generated.